lib/natalie/compiler/pass1.rb

```diff
@@ -844,6 +844,44 @@ def transform_class_variable_write_node(node, used:)
         instructions
       end
 
+      def transform_constant_and_write_node(node, used:)
+        instructions = [
+          PushSelfInstruction.new,
+          ConstFindInstruction.new(node.name, strict: false),
+          DupInstruction.new,
+          IfInstruction.new,
+          PopInstruction.new,
+          transform_expression(node.value, used: true),
+          DupInstruction.new,
+          PushSelfInstruction.new,
+          ConstSetInstruction.new(node.name),
+          ElseInstruction.new(:if),
+          EndInstruction.new(:if),
+        ]
+        instructions << PopInstruction.new unless used
+        instructions
+      end
+
+      def transform_constant_or_write_node(node, used:)
+        instructions = [
+          IsDefinedInstruction.new(type: 'constant'),
+          PushSelfInstruction.new,
+          ConstFindInstruction.new(node.name, strict: false),
+          EndInstruction.new(:is_defined),
```

Comment:
I wonder why this one needs the IsDefinedInstruction but the other transform does not 🤔

Reply:
https://kddnewton.com/2023/12/06/advent-of-prism-part-6#constants has some explanation for that